### PR TITLE
Add some bounds checks.

### PR DESCRIPTION
**Pull request by khuey**
Observed panics in the wild with malformed ELF.

### CODE REVIEW

**Review by nrc — APPROVED** [review]
Thansk!

**Comment by nrc** [comment]
@khuey do you want write access for this repo? I'm not sure my reviews add any value, so you could just push and not have to wait for me to get my shit together to approve

**Comment by khuey** [comment]
Sure.

**Pull request merged**
